cram/cram_encode.c

```diff
@@ -2530,6 +2543,8 @@ static int process_one_read(cram_fd *fd, cram_container *c,
     ref = c->ref;
     cr->flags       = bam_flag(b);
     cr->len         = bam_seq_len(b);
+    if (!bam_aux_get(b, "MD"))
+        MD = NULL;
```

Comment:
I think you need
```
else
    MD->l = 0;
```
here otherwise MD from all previous reads get carried over into the next one.

Reply:
Yes, this got moved and lost in the translation from dstring to kstring sadly.